src/main/scala/gestalt/Toolbox.scala

```diff
@@ -86,44 +86,33 @@ object flags {
 
 import flags._
 
-case class Modifiers[Tree](
-  flags: FlagSet = EmptyFlags,
-  privateWithin: String = "",   // can be empty or `this`
-  annotations: List[Tree] = Nil) {
-
-  def is(fs: FlagSet): Boolean = flags is fs
-  def is(fc: FlagSet, butNot: FlagSet): Boolean = flags.is(fc, butNot = butNot)
+trait Toolbox {
+  type Tree
+  type TypeTree <: Tree      // safety by construction -- implementation can have TypeTree = Tree
+  type Mods <: Modifiers
 
-  def | (fs: FlagSet): Modifiers[Tree] = withFlags(flags | fs)
-  def & (fs: FlagSet): Modifiers[Tree] = withFlags(flags & fs)
-  def &~(fs: FlagSet): Modifiers[Tree] = withFlags(flags &~ fs)
+  trait Modifiers {
+    def is(fs: FlagSet): Boolean
 
-  def withFlags(flags: FlagSet): Modifiers[Tree] =
-    copy(flags = flags)
+    def |(fs: FlagSet): Mods
 
-  def withAddedAnnotation(annot: Tree): Modifiers[Tree] =
-    withAnnotations(annotations :+ annot)
+    def &~(fs: FlagSet): Mods
 
-  def withAnnotations(annots: List[Tree]): Modifiers[Tree] =
-    copy(annotations = annots)
+    def withAddedAnnotation(annot: Tree): Mods
 
-  def withPrivateWithin(pw: String): Modifiers[Tree] =
-    copy(privateWithin = pw)
+    def hasAnnotations: Boolean
```

Comment:
Thanks @valdisxp1 . This is kind of intentional, usually programmers should only add new annotations, removing existing annotations seems strange. Let's be more conservative and add this feature when there's an imminent need for that.